### ISSUE TITLE
[Performance] How to achieve reported performance for DC model?

### ISSUE THREAD

**Issue by XiudingCai** [beanie00/Decision-ConvFormer#4]
Hello,

I’ve been using the official recommended script as follows:

```bash
# dc
python3 experiment.py --env hopper --dataset medium --model_type dc --K 8 --embed_dim 256 --learning_rate 0.0001
```

The results I obtained are as follows:

```
wandb: Run summary:
wandb: evaluation/target_36000_d4rl_score 63.74274
wandb: evaluation/target_36000_length_mean 633.32
wandb: evaluation/target_36000_return_mean 2054.28137
wandb: evaluation/target_36000_return_std 354.83809
wandb: evaluation/target_3600_d4rl_score 61.62763
wandb: evaluation/target_3600_length_mean 613.9
wandb: evaluation/target_3600_return_mean 1985.44362
wandb: evaluation/target_3600_return_std 282.30737
wandb: evaluation/target_72000_d4rl_score 64.96523
wandb: evaluation/target_72000_length_mean 645.09
wandb: evaluation/target_72000_return_mean 2094.06823
wandb: evaluation/target_72000_return_std 399.25478
wandb: evaluation/target_7200_d4rl_score 65.10807
wandb: evaluation/target_7200_length_mean 646.34
wandb: evaluation/target_7200_return_mean 2098.71693
wandb: evaluation/target_7200_return_std 393.40219
wandb: time/evaluation 394.93255
wandb: time/num_of_updates 100000
wandb: time/total 4892.80385
wandb: time/training 252.28102
wandb: training/action_error 0.05701
wandb: training/train_loss_mean 0.05709
wandb: training/train_loss_std 0.00507
```

The visualization logs on WandB are attached below.

![image](https://github.com/user-attachments/assets/996837d9-d38b-4228-8bc1-b42e0286b49b)

In the paper, Table 1 reports a performance of **92.5** for the DC model on the hopper-m. Could you kindly advise how I might be able to achieve this performance? Are there any specific parameters or adjustments I should make to the script?

Thank you in advance for your help!

**Comment by beanie00** [comment]
Hi XiudingCai, thank you for showing interest in our research. 
We evaluated the Return-Conditioned BC algorithms, including DC (ours), DT, ODT, and DS4, using the best results from 10 checkpoints. As shown in Figure 6 of the paper, we observed that performance tends to increase in DC as the normalized target RTG increases. Therefore, we evaluated the model using a target RTG of 72000 for the hopper task. Additionally, the results from the five seeds we tested were 97.1, 95.9, 89.4, 93.7, and 86.2, indicating some variability across seeds.
